### ISSUE TITLE
Blank page after undo, after setText

### ISSUE THREAD

**Issue by sbrown345** [firebase/firepad#242]
http://plnkr.co/edit/eHfBW8vHac7033PoNQwk?p=preview

Start with a firepad with some content in. Use setText to set some arbitary text, and then undo. The first undo state is a blank page, you have to undo again to get back to the previous text.

I'd expect the first undo to be the original content, not a blank page.

It happens with codemirror, ace, and the rich text editor. I've even tried using ace editor's 'setValue' with the same result. Once I disconnected firepad from the ace editor and used setValue it worked as expected: http://plnkr.co/edit/DoZ5RsyI5crU1ZbhAqYG?p=preview

**Comment by mikelehen** [comment]
Good point.  The fix is probably to wrap the setText implementation (the
setValue and insertText calls at least) here
<https://github.com/firebase/firepad/blob/master/lib/firepad.js#L185> in a
cm.operation() [see here
<https://github.com/firebase/firepad/blob/master/lib/firepad.js#L207> for
example].

**Comment by sbrown345** [comment]
I guess cm.operation()  is just for CodeMirror, is there an equivalent to cm.operation() for ace?
